test/mocha/10-restrictions.js

```diff
@@ -358,10 +393,63 @@ describe('restrictions', function() {
     let err;
     try {
       // try getting the removed restriction, this should throw a NotFoundError
-      getRestriction2 = await restrictions.get({
+      getRestriction2 = await restrictions.get({id});
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(getRestriction2);
+    should.exist(err);
+    err.name.should.equal('NotFoundError');
+    err.message.should.equal('Restriction not found.');
+  });
+
+  it('should remove a restriction from the database by id', async function() {
+    // create restriction
+    const id = await generateId();
+    const actualRestriction = await restrictions.insert({
+      restriction: {
+        id,
         zone: ZONES.ONE,
-        resource: RESOURCES.MANGO
-      });
+        resource: RESOURCES.MANGO,
+        method: 'limitOverDuration',
+        methodOptions: {
+          limit: 1,
+          duration: 'P30D'
+        }
+      }
+    });
+    const expectedRestriction = {
+      id,
+      zone: ZONES.ONE,
+      resource: RESOURCES.MANGO,
+      method: 'limitOverDuration',
+      methodOptions: {
+        limit: 1,
+        duration: 'P30D'
+      }
+    };
+    should.exist(actualRestriction);
+    should.exist(actualRestriction.meta);
+    should.exist(actualRestriction.restriction);
+    actualRestriction.restriction.should.deep.equal(expectedRestriction);
+
+    const getRestrictions = await restrictions.getAll({
+      zone: ZONES.ONE,
+      resource: RESOURCES.MANGO
+    });
+    const getRestriction = getRestrictions.restrictions[0];
+    should.exist(getRestriction);
+    should.exist(getRestriction.meta);
+    should.exist(getRestriction.restriction);
+    getRestriction.restriction.should.deep.equal(expectedRestriction);
+
+    // remove the restriction
+    await restrictions.removeById({id});
+    let getRestriction2;
```

Comment:
variables with a verb like `get` in them are typically reserved for functions.  I suggest: `restrictionResult2` or `restrictionRecord2`.